doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -659,168 +734,476 @@ insert in a concise way an Ltac2 variable of type :n:`constr` into a Coq term.
+         * - Pattern syntax
+           - Example pattern
 
-Scopes
-~~~~~~
+         * - :n:`@name : @ltac2_match_pattern`
+           - `n : ?t`
 
-A scope is a name given to a grammar entry used to produce some Ltac2 expression
-at parsing time. Scopes are described using a form of S-expression.
+         * - :n:`@name := @match_pattern__binder`
+           - `n := ?b`
 
-.. prodn::
-   ltac2_scope ::= {| @string | @int | @lident ({+, @ltac2_scope}) }
+         * - :n:`@name := @term__binder : @type`
+           - `n := ?b : ?t`
 
-A few scopes contain antiquotation features. For the sake of uniformity, all
-antiquotations are introduced by the syntax :n:`$@lident`.
+         * - :n:`@name := [ @match_pattern__binder ] : @ltac2_match_pattern`
+           - `n := [ ?b ] : ?t`
 
-The following scopes are built-in.
+         :token:`name` can't have a `?`.  Note that the last two forms are equivalent except that:
 
-- :n:`constr`:
+         - if the `:` in the third form has been bound to something else in a notation, you must use the fourth form.
+           Note that cmd:`Require Import` `ssreflect` loads a notation that does this.
+         - a :n:`@term__binder` such as `[ ?l ]` (e.g., denoting a singleton list after
+           :cmd:`Import` `ListNotations`) must be parenthesized or, for the fourth form,
+           use double brackets: `[ [ ?l ] ]`.
 
-  + parses :n:`c = @term` and produces :n:`constr:(c)`
+      If there are multiple :token:`gmatch_hyp_pattern`\s in a branch, there may be multiple ways to match them to hypotheses.
+      For :tacn:`match! goal` and :tacn:`multi_match! goal`, if the evaluation of the :token:`ltac2_expr` fails,
+      matching will continue with the next hypothesis combination.  When those are exhausted,
+      the next alternative from any `context` constructs in the :token:`ltac2_match_pattern`\s is tried and then,
+      when the context alternatives are exhausted, the next branch is tried.
+      :ref:`Example<ltac2_match_goal_multiple_hyps_ex>`.
 
-  This scope can be parameterized by a list of delimiting keys of notation
-  scopes (as described in :ref:`LocalInterpretationRulesForNotations`),
-  describing how to interpret the parsed term. For instance, :n:`constr(A, B)`
-  parses :n:`c = @term` and produces :n:`constr:(c%A%B)`.
+   `reverse`
+      Hypothesis matching for :token:`gmatch_hyp_pattern`\s normally begins by matching them from left to right,
+      to hypotheses, last to first.  Specifying `reverse` begins matching in the reverse order, from
+      first to last.  :ref:`Normal<ltac2_match_goal_hyps_ex>` and :ref:`reverse<ltac2_match_goal_hyps_rev_ex>` examples.
 
-- :n:`ident`:
+   :n:`|- @ltac2_match_pattern`
+      A pattern to match with the current goal
 
-  + parses :n:`id = @ident` and produces :n:`ident:(id)`
-  + parses :n:`$(x = @ident)` and produces the variable :n:`x`
+   Note that unlike Ltac, only lowercase identifiers are valid as Ltac2
```

Comment:
```suggestion
   Note that unlike Ltac1, only lowercase identifiers are valid as Ltac2
```

Note also that this is duplicated text; is that intentional?

Reply:
Yes, intentional.  One is in match over terms, the other is in match over goals.  Readers should not have to read both to glean this fact, so some repetition is beneficial.  A long repetition should be replaced with shared and hyperlinked text.